### PR TITLE
docs: vee-validate plugin documentation

### PR DESCRIPTION
**Pull request by logaretm**
This PR adds the documentation of the [vee-validate plugin](https://github.com/formvuelate/formvuelate-plugin-vee-validate).

There is a simple `TODO` left which is to provide a link to a live example once everything is released, so you may not want to merge this until then.

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -915,6 +917,309 @@ const SchemaFormWithPlugin = SchemaFormFactory([
 ])
 ```
 
+### Vee-Validate Plugin
+
+[Lookup Plugin's Repo](https://github.com/vuelidate/formvuelate-plugin-vee-validate).
```
Comment (marina-mosti) [inline index.md]:
```suggestion
[Vee-Validate Plugin's Repo](https://github.com/vuelidate/formvuelate-plugin-vee-validate).
```

File: index.md [hunk 2]
```diff
@@ -915,6 +917,309 @@ const SchemaFormWithPlugin = SchemaFormFactory([
...
+{
+  errorMessage, // The first error message for that field
+  errors, // All error messages for that field
+  meta, // A field meta object
+  setDirty, // Sets the meta `dirty` flag
+  setTouched, // Sets the meta `touched` flag
+}
+```
+
+You can opt-in to any of these properties or to the entire `validation` object, here is a simple `FormText` component that accepts the `validation` object as a prop:
```
Comment (marina-mosti) [inline index.md]:
```suggestion
You can opt-in to any of these properties or to the entire `validation` object. Here is an example `FormText` component that accepts the `validation` object as a prop:
```

File: index.md [hunk 3]
```diff
@@ -915,6 +917,309 @@ const SchemaFormWithPlugin = SchemaFormFactory([
...
+  methods: {
+    update (value) {
+      this.$emit('update:modelValue', value)
+    }
+  }
+}
+</script>
+```
+
+Whenever the `modelValue` is updated the field will be validated immediately. If you don't want that and want your validations to be less aggressive, then you can show the error message when the field is `touched` which is set after the field is blurred and you can set it using `setTouched` on the validation object.
```
Comment (marina-mosti) [inline index.md]:
```suggestion
Whenever the `modelValue` is updated the field will be validated immediately. 

If we want the validations not to be immediate, or lazy, we can show the error message when the field is `touched`. A field is considered `touched` after the field loses focus. 

In this case, we set it using the validation object's `setTouched` method as shown in the following example.
```

File: index.md [hunk 4]
```diff
@@ -915,6 +917,309 @@ const SchemaFormWithPlugin = SchemaFormFactory([
...
+      this.validation.setTouched(true);
+    }
+  }
+}
+</script>
+```
+
+#### Configuration
+
+The `VeeValidatePlugin` takes one parameter which is a configuration object, here are the properties that you can use.
```
Comment (marina-mosti) [inline index.md]:
```suggestion
The `VeeValidatePlugin` accepts one parameter, a configuration object. Let's look at the properties that we can use.
```

File: index.md [hunk 5]
```diff
@@ -915,6 +917,309 @@ const SchemaFormWithPlugin = SchemaFormFactory([
...
+
+##### mapProps
+
+If you are using 3rd party components and cannot modify their definition to accept the `validation` object, you can use the `mapProps` configuration to map the `validation` object to another property or multiple properties that are accepted by your component.
```
Comment (marina-mosti) [inline index.md:1070]:
```suggestion

::: warning Important
This refers to Vee-validate's `mapProps`, and not to the `mapProps` property exposed by `LookupPlugin`. 
:::

If you are using 3rd party components and cannot modify their definition to accept the `validation` object, you can use the `mapProps` configuration to map the `validation` object to another property or multiple properties that are accepted by your component.
```

**Review by marina-mosti — COMMENTED** [review]
Fantastic job on the docs @logaretm !
Left a few tweaks and suggestions :)

**Comment by logaretm** [comment]
Thank you, I committed the suggestions.

**Review by marina-mosti — APPROVED** [review]
Looks great :)

**Comment by logaretm** [comment]
I removed the demo bit, should be ready.

**Pull request merged**